src/v2/Apps/Consign/Components/GetPriceEstimate/ConsignPriceEstimateContext.tsx

```diff
@@ -1,32 +1,76 @@
-import React, { useContext, useReducer } from "react"
+import React, { Dispatch, useContext, useReducer } from "react"
 import { createContext } from "react"
-import { fetchQuery, graphql } from "react-relay"
+import { Environment, fetchQuery, graphql } from "react-relay"
 import { useSystemContext } from "v2/Artsy"
 
 import { ConsignPriceEstimateContext_SearchConnection_Query } from "v2/__generated__/ConsignPriceEstimateContext_SearchConnection_Query.graphql"
 import { ConsignPriceEstimateContext_ArtistInsights_Query } from "v2/__generated__/ConsignPriceEstimateContext_ArtistInsights_Query.graphql"
 
 interface PriceEstimateContextProps {
-  artistInsights?: object
+  artistInsights?: ArtistInsights
+  fetchArtistInsights?: (suggestion: Suggestion) => void
   fetchSuggestions?: (searchQuery: string) => void
+  isFetching?: boolean
   searchQuery?: string
-  selectSuggestion?: (suggestion: any) => void
+  selectSuggestion?: (suggestion: Suggestion) => void
+  selectedSuggestion?: Suggestion
+  setFetching?: (isFetching: boolean) => void
   setSearchQuery?: (searchQuery: string) => void
-  suggestions?: any[]
+  suggestions?: Suggestions
 }
 
-const initialState = {
+type ArtistInsights = ConsignPriceEstimateContext_ArtistInsights_Query["response"]
+type Suggestions = ConsignPriceEstimateContext_SearchConnection_Query["response"]["searchConnection"]["edges"]
+type Suggestion = Suggestions[0]
+
+type State = Pick<
```

Comment:
Cool, didn't know about `Pick` 💯

Reply:
Lots of goodies in here: https://www.typescriptlang.org/docs/handbook/utility-types.html